src/actions/channel_categories.ts

```diff
@@ -186,12 +190,21 @@ export function moveChannelToCategory(categoryId: string, channelId: string, new
             });
         }
 
-        let data;
+        const result = dispatch({
+            type: ChannelCategoryTypes.RECEIVED_CATEGORIES,
+            data: categories,
+        });
+
         try {
-            data = await Client4.updateChannelCategories(currentUserId, category.team_id, categories);
+            await Client4.updateChannelCategories(currentUserId, category.team_id, categories);
         } catch (error) {
             forceLogoutIfNecessary(error, dispatch, getState);
             dispatch(logError(error));
+
+            dispatch({
+                type: ChannelCategoryTypes.RECEIVED_CATEGORIES,
+                data: originalCategories,
```

Comment:
Relating to above, this should also include `originalCategory` since that may have been changed as well

Reply:
Not sure I understand, but here we need to Null-ify the optimistic update, so we are trying to make it so as if nothing happened. We just have the previous state and that should be enough?

Reply:
The problem is that `sourceCategories` only includes the original version of `category` and not `originalCategory`, so we're only reverting `category` if something fails

Reply:
That was the idea, we optimistically dispatch the changed ```category``` and then if that fails, we roll back to the original state. I think I might be missing something here, given what you are saying?

Reply:
I think I understood, we need to roll back both categories involved in the change, original and source category. Is that correct?

Reply:
Maybe we should rename them 'source' and 'target' since it was a bit confusing?

Reply:
Yeah, I'd be up for renaming them like that. I think the renaming would go
- category -> targetCategory
- originalCategory -> sourceCategory
- categories stays the same
- sourceCategories -> originalCategories